src/cml.js

```diff
@@ -402,7 +402,13 @@ class CML {
   }
 
   async startRunner(opts = {}) {
-    return await this.getDriver().startRunner(opts);
+    const env = {};
+    const sensitive =
+      ['CML_RUNNER_SENSITIVE_ENV'] +
```

Comment:
Would a whitelist of environment variables to propagate be a more robust solution here?

Reply:
Indeed, that's what I originally proposed:
* https://github.com/iterative/cml/issues/721#issuecomment-923797961
* https://iterativeai.slack.com/archives/C01900GSB4J/p1631893922227800
* https://iterativeai.slack.com/archives/C01900GSB4J/p1632223956257700

Although that would require launching the runner process from a shell, to provide the bare minimum variables like e.g. `PATH` and any custom NVIDIA values.

Reply:
Again 

> I am in favor of setting up the CI agents as their own service that the runner monitors so that their parent is systemd and not cml, thus avoiding the env leaks.

would also solve it, would it not?

Reply:
> _would also solve it, would it not?_

Except for Kubernetes

Reply:
> would also solve it, would it not?

It depends how you would like to solve this. Cloud agnostic should be the way that way CML is secure in every environment. I might want to just launch CML on my own

Reply:
```js
const sensitive = ['CML_RUNNER_SENSITIVE_ENV'] + process.env.CML_RUNNER_SENSITIVE_ENV.split(':');
for (const variable in process.env)
      if (!sensitive.includes(variable)) env[variable]
```

This does not sounds to me correct. Have you tried this?

in JS `+` concatenates strings so I think that this is calling the toSting() incarnation of them giving you 
```
CML_RUNNER_SENSITIVE_ENVone,two,three 
```
for process.env.CML_RUNNER_SENSITIVE_ENV = ['one:two:three']

Reply:
It works, but it's definitely not correct. 🤦🏼 Thanks!

Reply:
JavaScript blunder fixed with https://github.com/iterative/cml/pull/1164/commits/e29b8aad00437c8174c3f4ec49ea7944557f1c06 (commit comment worth watching)

Reply:
> _Would a whitelist of environment variables to propagate be a more robust solution here?_

Better solutions are possible, but also more complicated. To be addressed in the future.